File: LENGUAJES/PYTHON/Ejercicio7.py
#FUNCIONES

def obtener_nombre():
    while True:
        nombre = input("Ingrese su nombre: ").strip()
        if nombre:
            return nombre
        else:
            print("El nombre no puede estar vacío. Inténtalo de nuevo.")

def saludo(nombre):
    print(f"Hola {nombre}, bienvenido/a!")

def main():
    while True:
        nombre = obtener_nombre()
        saludo(nombre)
        
        continuar = input("¿Deseas saludar a otra persona? (s/n): ").strip().lower()
        if continuar != 's':
            print("Gracias por usar el programa. ¡Hasta luego!")
            break
main()

File: LENGUAJES/PHP/Ejercicio7.php
<?php
// FUNCIONES

function obtenerNombre() {
    while (true) {
        echo "Ingrese su nombre: ";
        $nombre = trim(fgets(STDIN));
        
        if (!empty($nombre)) {
            return $nombre;
        } else {
            echo "El nombre no puede estar vacío. Inténtalo de nuevo.\n";
        }
    }
}

function saludo($nombre) {
    echo "Hola $nombre, bienvenido/a!\n";
}

function main() {
    while (true) {
        $nombre = obtenerNombre();
        saludo($nombre);
        
        echo "¿Deseas saludar a otra persona? (s/n): ";
        $continuar = trim(fgets(STDIN));
        
        if (strtolower($continuar) !== 's') {
            echo "Gracias por usar el programa. ¡Hasta luego!\n";
            break;
        }
    }
}
main();
?>